filters/parser.go

```diff
@@ -41,7 +41,7 @@ The formal grammar is as follows:
 selectors := selector ("," selector)*
 selector  := fieldpath (operator value)
 fieldpath := field ('.' field)*
-field     := quoted | [A-Za-z] [A-Za-z0-9_]+
+field     := quoted | [A-Za-z-] [A-Za-z0-9_]+
```

Comment:
Would a field start from `-`?

Reply:
I think 'dash' that separates two words is taken as a field/value. Also, [A-z0-9-+()]+ is a hyphen(-) is accepted in regex. Here an example [-A-z0-9]+, [a-Za-9-]+, [a-z-0-9]+, and [a-z-0-9]+ are all similar.

Reply:
Before your change, `hello`, `hello1` and `hello_1` are all valid fields, but `1hello` and `_hello` are not.

The comment after your change indicates, `_hello` are still invalid but `-hello` is valid. Is that what you want to propose?